staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -187,6 +187,8 @@ type Instances interface {
 	InstanceExistsByProviderID(ctx context.Context, providerID string) (bool, error)
 	// InstanceShutdownByProviderID returns true if the instance is shutdown in cloudprovider
 	InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error)
+	// InstanceMetadataByProviderID returns the instance's metadata which contains its type and nodeAddresses.
```

Comment:
nit: I would remove "which contains..." part. We may add something to it later and this will become outdated.

Reply:
Addressed.